src/transformers/models/auto/modeling_auto.py

```diff
@@ -1002,6 +1002,21 @@
     ]
 )
 
+MODEL_FOR_TEXT_TO_SPECTROGRAM_MAPPING_NAMES = OrderedDict(
+    [
+        # Model for Text-To-Spectrogram mapping
+        ("speecht5", "SpeechT5ForTextToSpeech"),
+    ]
+)
+
+MODEL_FOR_TEXT_TO_WAVEFORM_MAPPING_NAMES = OrderedDict(
+    [
+        # Model for Text-To-Waveform mapping
+        ("bark", "BarkModel"),
+        ("musicgen", "MusicgenForConditionalGeneration"),
+    ]
+)
+
```

Comment:
There are currently two types of TTS models:

-  **Text-to-waveform models** which directly generates a waveform
-  **Text-to-spectrogram models** which generates a spectrogram that needs to be converted to a waveform with a vocoder.

Reply:
We'll need to make sure these lists are updated when new models are added

Reply:
For sure!